### PR TITLE
feat: YDB JOOQ extension doc

### PR DESCRIPTION
**Pull request by KirillKurdyukov**
### Changelog category 

* Documentation (changelog entry is not required)

### CODE REVIEW

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# JOOQ Extension for use with {{ ydb-short-name }}
+
+This guide is intended for use with [JOOQ](https://www.jooq.org/) and {{ ydb-short-name }}.
+
+JOOQ is a Java library that allows you to create type-safe SQL queries by generating Java classes from a database schema and using convenient query builders.
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:5]:
```suggestion
# JOOQ extension for {{ ydb-short-name }}

This guide explains how to use [JOOQ](https://www.jooq.org/) with {{ ydb-short-name }}.

JOOQ is a Java library that allows you to create type-safe SQL queries by generating Java classes from a database schema and providing convenient query builders.
```

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+## Generating Java Classes {#generated-java-classes}
+
+You can generate Java classes using any tools provided on the [official JOOQ website](https://www.jooq.org/doc/latest/manual/code-generation/codegen-configuration/). Two dependencies are required: {{ ydb-short-name }} JDBC driver and JOOQ extension for {{ ydb-short-name }}, as well as two parameters:
+
+- `database.name`: `tech.ydb.jooq.codegen.YdbDatabase` (mandatory setting)
+- `strategy.name`: `tech.ydb.jooq.codegen.YdbGeneratorStrategy` (recommended setting)
+
+Let's look at an example of the `maven` plugin:
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:14]:
```suggestion
You can generate Java classes using any of the tools provided on the [official JOOQ website](https://www.jooq.org/doc/latest/manual/code-generation/codegen-configuration/). Two dependencies are required: the [{{ ydb-short-name }} JDBC driver](https://github.com/ydb-platform/ydb-jdbc-driver) and the JOOQ extension for {{ ydb-short-name }}, along with two parameters:

- `database.name`: `tech.ydb.jooq.codegen.YdbDatabase` (mandatory setting)
- `strategy.name`: `tech.ydb.jooq.codegen.YdbGeneratorStrategy` (recommended setting)

An example using the `maven` plugin:
```

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            <target>
+                <packageName>ydb</packageName>
+                <directory>./src/main/java</directory>
+            </target>
+        </generator>
+    </configuration>
+</plugin>
+```
+
+Example of generated classes from [YQL tutorial](../../dev/yql-tutorial/create_demo_tables.md):
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:63]: It'd be helpful to post the full classes somewhere on GitHub and give a link to them here.

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            <driver>tech.ydb.jdbc.YdbDriver</driver>
+            <url>jdbc:ydb:grpc://localhost:2136/local</url>
+        </jdbc>
+        <generator>
+            <strategy>
+                <name>tech.ydb.jooq.codegen.YdbGeneratorStrategy</name>
+            </strategy>
+            <database>
+                <name>tech.ydb.jooq.codegen.YdbDatabase</name>
+                <!-- исключение системных таблицы -->
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:51]:
```suggestion
                <!-- excluding system tables -->
```

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        <artifactId>ydb-jdbc-driver</artifactId>
+        <version>${ydb.jdbc.version}</version>
+    </dependency>
+
+    <dependency>
+        <groupId>tech.ydb.dialects</groupId>
+        <artifactId>spring-data-jdbc-ydb</artifactId>
+        <version>${spring.data.jdbc.ydb}</version> 
+    </dependency>
+    ```
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:105]: There was a similar example in lines 16-60: why do you give a complete example first and then repeat a small part of it?

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        <groupId>tech.ydb.jdbc</groupId>
+        <artifactId>ydb-jdbc-driver</artifactId>
+        <version>${ydb.jdbc.version}</version>
+    </dependency>
+
+    <dependency>
+        <groupId>tech.ydb.dialects</groupId>
+        <artifactId>spring-data-jdbc-ydb</artifactId>
+        <version>${spring.data.jdbc.ydb}</version> 
+    </dependency>
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:104]: This looks like an incorrect copy-paste from a neighboring Spring JDBC article.

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    ```
+
+- Gradle
+
+    ```groovy
+    dependencies {
+        // Set actual versions
+        implementation "tech.ydb.dialects:spring-data-jdbc-ydb:$ydbDialectVersion"
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:112]: also Spring JDBC

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+        implementation "tech.ydb.jdbc:ydb-jdbc-driver:$ydbJdbcVersion"
+    }
+    ```
+
+{% endlist %}
+
+To obtain a `YdbDSLContext` (an extension of `org.jooq.DSLContext`), use the `tech.ydb.jooq.YDB` class. For example:
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:119]:
```suggestion
To obtain a `YdbDSLContext` class instance (an extension of `org.jooq.DSLContext`), use the `tech.ydb.jooq.YDB` class. For example:
```

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+```java
+String url = "jdbc:ydb:<grpc/grpcs>://<host>:<2135/2136>/path/to/database[?saFile=file:~/sa_key.json]";
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:122]: An endpoint has already been specified above at line 43; why is it needed two times?

Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:122]: `<...>` in our documentation is usually used just to name a placeholder (not to list all options), then below the example, each placeholder needs to be explained.

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    @Override
+    public YdbDSLContextImpl dslContext(org.jooq.Configuration configuration) {
+        return YdbDSLContextImpl(configuration);
+    }
+}
+```
+
+```properties
+spring.datasource.driver-class-name=tech.ydb.jdbc.YdbDriver
+spring.datasource.url=jdbc:ydb:<grpc/grpcs>://<host>:<2135/2136>/path/to/database[?saFile=file:~/sa_key.json]
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:152]: If this indeed is intended to be used inside the Spring Framework, let's make separate examples for each scenario and explicitly explain what options are available.

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+```
+
+`YdbDSLContext` is ready to use. Let's look at examples of specific YQL queries:
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:155]:
```suggestion
`YdbDSLContext` is ready to use.

 ## Examples of specific YQL queries
```

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+The [`UPSERT`](../../yql/reference/syntax/upsert_into.md) command:
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:157]: Let's add a YQL query corresponding to each of these snippets.

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+
+```java
+public void replace(YdbDSLContext context) {
+    ydbDSLContext.replaceInto(EPISODES)
+            .set(record)
+            .execute();
+}
+```
+
+To specify an index in the VIEW statement, you must use the `useIndex` method on `Table`:
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:177]:
```suggestion
To specify a [secondary index](../../concepts/secondary_indexes.md) in the `VIEW` statement, you must use the `useIndex` method on `Table`:
```

**Review by blinkov — CHANGES_REQUESTED** [review]

File: ydb/docs/en/core/integrations/orm/jooq.md (hunk 9 above)
Comment (KirillKurdyukov) [reply ydb/docs/en/core/integrations/orm/jooq.md:122]: Первый URL для генерации Java классов с помощью мавен плагина. Это обычно делается через докер, накатив предварительно схему

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 14]
```diff
@@ -0,0 +1,202 @@
...
+```java
+// generated SQL:
+// select `series`.`series_id`, `series`.`title`, `series`.`series_info`, `series`.`release_date` 
+// from `series` view `title_name` where `series`.`title` = ?
+var record = ydbDSLContext.selectFrom(SERIES.useIndex(Indexes.TITLE_NAME.name))
+        .where(SERIES.TITLE.eq(title))
+        .fetchOne();
+```
+
+In all other respects, the {{ ydb-short-name }} dialect follows the JOOQ documentation.
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:180]:
```suggestion
In all other respects, the {{ ydb-short-name }} dialect follows the [JOOQ documentation](https://www.jooq.org/doc/latest/manual/).
```

File: ydb/docs/en/core/integrations/orm/jooq.md [hunk 15]
```diff
@@ -0,0 +1,202 @@
...
+    }
+}
+```
+
+```properties
+spring.datasource.driver-class-name=tech.ydb.jdbc.YdbDriver
+spring.datasource.url=jdbc:ydb:<schema>://<host>:<port>/path/to/database[?saFile=file:~/sa_key.json]
+```
+
+An example of a simple Spring Boot application can be found [here](https://github.com/ydb-platform/ydb-java-examples/tree/master/jdbc/spring-jooq).
```
Comment (blinkov) [inline ydb/docs/en/core/integrations/orm/jooq.md:202]:
```suggestion
A complete example of a simple Spring Boot application can be found [here](https://github.com/ydb-platform/ydb-java-examples/tree/master/jdbc/spring-jooq).
```

File: ydb/docs/ru/core/integrations/orm/jooq.md [hunk 16]
```diff
@@ -0,0 +1,202 @@
+# Расширение JOOQ для использования с {{ ydb-short-name }}
+
+Это руководство предназначено для использования [JOOQ](https://www.jooq.org/) с {{ ydb-short-name }}.
+
+JOOQ — это библиотека для Java, которая позволяет создавать типобезопасные SQL-запросы путем генерации Java-классов из схемы базы данных и использования удобных конструкторов запросов.
```
Comment (blinkov) [inline ydb/docs/ru/core/integrations/orm/jooq.md:5]:
```suggestion
# Расширение JOOQ для использования с {{ ydb-short-name }}

Это руководство предназначено для использования [JOOQ](https://www.jooq.org/) с {{ ydb-short-name }}.

JOOQ — это библиотека для Java, которая позволяет создавать типобезопасные SQL-запросы путём генерации Java-классов из схемы базы данных и использования удобных конструкторов запросов.
```

File: ydb/docs/ru/core/integrations/orm/jooq.md [hunk 17]
```diff
@@ -0,0 +1,202 @@
...
+
+## Генерация Java классов
+
+Генерировать Java классы можно любыми инструментами представленными на [официальном сайте JOOQ](https://www.jooq.org/doc/latest/manual/code-generation/codegen-configuration/), используя две зависимости {{ ydb-short-name }} JDBC Driver и расширение JOOQ для {{ ydb-short-name }}. Также требуется указать два параметра:
+
+- `database.name`: `tech.ydb.jooq.codegen.YdbDatabase` (обязательная настройка)
+- `strategy.name`: `tech.ydb.jooq.codegen.YdbGeneratorStrategy` (рекомендуется к использованию)
```
Comment (blinkov) [inline ydb/docs/ru/core/integrations/orm/jooq.md:12]:
```suggestion
## Генерация Java-классов

Генерировать Java-классы можно с помощью любых инструментов, представленных на [официальном сайте JOOQ](https://www.jooq.org/doc/latest/manual/code-generation/codegen-configuration/), используя две зависимости: {{ ydb-short-name }} JDBC Driver и расширение JOOQ для {{ ydb-short-name }}. Также необходимо указать два параметра:

- `database.name`: `tech.ydb.jooq.codegen.YdbDatabase` (обязательная настройка)
- `strategy.name`: `tech.ydb.jooq.codegen.YdbGeneratorStrategy` (рекомендуется к использованию)
```

File: ydb/docs/ru/core/integrations/orm/jooq.md [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+            <target>
+                <packageName>ydb</packageName>
+                <directory>./src/main/java</directory>
+            </target>
+        </generator>
+    </configuration>
+</plugin>
+```
+
+Пример сгенерированных классов из [туториал по YQL](../../dev/yql-tutorial/create_demo_tables.md) (сгенерированные классы доступны по [ссылке](https://github.com/ydb-platform/ydb-java-examples/tree/master/jdbc/spring-jooq/src/main/java/ydb/default_schema)):
```
Comment (blinkov) [inline ydb/docs/ru/core/integrations/orm/jooq.md:63]:
```suggestion
Пример сгенерированных классов из [туториала по YQL](../../dev/yql-tutorial/create_demo_tables.md) (их полный код [доступен на GitHub](https://github.com/ydb-platform/ydb-java-examples/tree/master/jdbc/spring-jooq/src/main/java/ydb/default_schema)):
```

File: ydb/docs/ru/core/integrations/orm/jooq.md [hunk 19]
```diff
@@ -0,0 +1,202 @@
...
+ydb/default_schema/Keys.java
+ydb/default_schema/Tables.java
+ydb/default_schema/DefaultSchema.java
+```
+
+## Использование
+
+Для интеграции {{ ydb-short-name }} с JOOQ в ваш проект потребуется добавить две зависимости: {{ ydb-short-name }} JDBC Driver и расширение JOOQ для {{ ydb-short-name }}.
+
+Примеры для различных систем сборки:
```
Comment (blinkov) [inline ydb/docs/ru/core/integrations/orm/jooq.md:86]:
```suggestion
## Использование

Для интеграции {{ ydb-short-name }} с JOOQ в ваш проект потребуется добавить две зависимости: {{ ydb-short-name }} JDBC Driver и расширение JOOQ для {{ ydb-short-name }}.

Примеры для различных систем сборки:
```

File: ydb/docs/ru/core/integrations/orm/jooq.md [hunk 20]
```diff
@@ -0,0 +1,202 @@
...
+try(CloseableYdbDSLContext dsl = YDB.using(url)) {
+    // ...
+}
+```
+
+`YdbDSLContext` готов к использованию.
+
+## YQL команды
+
+Из синтаксиса YQL доступны следующие команды в `YdbDSLContext`:
```
Comment (blinkov) [inline ydb/docs/ru/core/integrations/orm/jooq.md:141]:
```suggestion
В `YdbDSLContext` доступны следующие команды, специфичные для синтаксиса YQL:
```

File: ydb/docs/ru/core/integrations/orm/jooq.md [hunk 21]
```diff
@@ -0,0 +1,202 @@
...
+```java
+// generated SQL:
+// select `series`.`series_id`, `series`.`title`, `series`.`series_info`, `series`.`release_date` 
+// from `series` view `title_name` where `series`.`title` = ?
+var record = ydbDSLContext.selectFrom(SERIES.useIndex(Indexes.TITLE_NAME.name))
+        .where(SERIES.TITLE.eq(title))
+        .fetchOne();
+```
+
+В остальном диалект {{ ydb-short-name }} следует документации JOOQ.
```
Comment (blinkov) [inline ydb/docs/ru/core/integrations/orm/jooq.md:180]:
```suggestion
В остальном диалект {{ ydb-short-name }} соответствует [документации JOOQ](https://www.jooq.org/doc/latest/manual/).
```

File: ydb/docs/ru/core/integrations/orm/jooq.md [hunk 22]
```diff
@@ -0,0 +1,202 @@
...
+    }
+}
+```
+
+```properties
+spring.datasource.driver-class-name=tech.ydb.jdbc.YdbDriver
+spring.datasource.url=jdbc:ydb:<schema>://<host>:<port>/path/to/database[?saFile=file:~/sa_key.json]
+```
+
+Пример простого приложения Spring Boot можно найти по [ссылке](https://github.com/ydb-platform/ydb-java-examples/tree/master/jdbc/spring-jooq).
```
Comment (blinkov) [inline ydb/docs/ru/core/integrations/orm/jooq.md:202]:
```suggestion
Полный пример простого приложения Spring Boot можно найти [на GitHub](https://github.com/ydb-platform/ydb-java-examples/tree/master/jdbc/spring-jooq).
```

**Review by blinkov — DISMISSED** [review]

**Review by fomichev3000 — APPROVED** [review]

**Pull request merged**